test/test_exceptions.py

```diff
@@ -47,118 +49,176 @@ def test_gate_error_str_and_repr():
     str_expected = ("IonQGateError(\"gate 'a gate' is not supported on the 'b set' IonQ backends."
                     " Please use the qiskit.transpile method, manually rewrite to remove the gate,"
                     " or change the gateset selection as appropriate.\")"
-                   )
+                    )
     repr_expected = "IonQGateError(gate_name='a gate', gateset='b set')"
     assert str(err) == str_expected
     assert repr(err) == repr_expected
 
 
 def test_api_error():
     """Test that IonQAPIError has specific instance attributes."""
-    err = exceptions.IonQAPIError("an error", 500, "internal_error")
+    err = exceptions.IonQAPIError(
+        message="an error",
+        status_code=500,
+        headers={"Content-Type": "text/html"},
+        body="<!doctype html><html><body>Hello IonQ!</body></html>",
+        error_type="internal_error"
+    )
     assert err.message == "an error"
     assert err.status_code == 500
+    assert err.headers == {"Content-Type": "text/html"}
+    assert err.body == "<!doctype html><html><body>Hello IonQ!</body></html>"
     assert err.error_type == "internal_error"
 
 
 def test_api_error_str_and_repr():
     """Test that IonQAPIError has a str/repr that includes args."""
-    err = exceptions.IonQAPIError("an error", 500, "internal_error")
-    expected = "IonQAPIError(message='an error',status_code=500,error_type='internal_error')"
+    err = exceptions.IonQAPIError(
+        message="an error",
+        status_code=500,
+        headers={"Content-Type": "text/html"},
+        body="<!doctype html><html><body>Hello IonQ!</body></html>",
+        error_type="internal_error"
+    )
+    expected = ("IonQAPIError(message='an error',"
+                "status_code=500,"
+                "headers={'Content-Type': 'text/html'},"
+                "body=<!doctype html><html><body>Hello IonQ!</body></html>,"
+                "error_type='internal_error')")
     assert str(err) == expected
     assert repr(err) == repr(expected)
 
 
 def test_api_error_from_response():
     """Test that IonQAPIError can be made directly from a response JSON dict."""
-    fake_response = mock.MagicMock()
-    fake_response.json = mock.MagicMock(
+    # Create a response object
+    response = requests.Response()
+    response.status_code = 500
+    response.headers = {"Content-Type": "text/html"}
+    response._content = b"<!doctype html><html><body>Hello IonQ!</body></html>"
+    # Mock the json method of the response object
+    response.json = mock.MagicMock(
```

Comment:
you might be able to avoid mocking this if you put a json string into _content (though that seems a bit odd too. It's test only code so I don't mind much, but we could use https://github.com/getsentry/responses if you're interested in trying it out)